pkg/migrations/mls/20240829001344_serial-ids.up.sql

```diff
@@ -0,0 +1,44 @@
+CREATE FUNCTION insert_group_message(group_id BYTEA, data BYTEA, group_id_data_hash BYTEA)
+	RETURNS SETOF group_messages
+	AS $$
+BEGIN
+	-- Ensures that the generated sequence ID matches the insertion order
+	-- Only released at the end of the enclosing transaction - beware if called within a long transaction
+	PERFORM
+		pg_advisory_xact_lock(hashtext('group_messages_sequence'), hashtext(encode(group_id, 'hex')));
```

Comment:
We could cut down on the number of hashes by concatenating the two strings before hashing if we're concerned about perf

Reply:
Another good option if perf becomes an issue. Separated it for this implementation to reduce collisions (especially against other tables)